src/auth/auth.ts

```diff
@@ -0,0 +1,64 @@
+import Client from '../api_client/client';
+import { GenericResponse } from '../core/resreq';
+import { KwilSigner } from '../core/kwilSigner';
+import { AuthSuccess, AuthenticatedBody, LogoutResponse, composeAuthMsg, removeTrailingSlash, verifyAuthProperties } from '../core/auth';
+import { BytesEncodingStatus, EnvironmentType } from '../core/enums';
+import { objects } from '../utils/objects';
+import { executeSign } from '../core/signature';
+import { stringToBytes } from '../utils/serial';
+import { bytesToBase64 } from '../utils/base64';
+
+export class Auth<T extends EnvironmentType> {
+  private client: Client;
+  private kwilProvider: string;
+  private chainId: string;
+
+  public constructor(client: Client, kwilProvider: string, chainId: string) {
+    this.client = client;
+    this.kwilProvider = kwilProvider;
+    this.chainId = chainId;
+  }
+
+  /**
+   * Authenticates a user with the Kwil Gateway (KGW). This is required to execute mustsign view actions.
```

Comment:
The comment is outdated, 'mustsign' action is no longer supported

Reply:
Good catch